R/viz.R

```diff
@@ -2354,18 +2356,25 @@ prettyForestPlot = function(maf,
       dplyr::rename("Tumor_Sample_Barcode"="sample_id")
   }else if(!missing(maf)){
     if(separate_hotspots){
-        if(!"hot_spot" %in% colnames(maf))
-          stop("No \"hot_spot\" column in maf file. Annotate your maf file with GAMBLR::annotate_hot_spots() first. ")
+      if(!"hot_spot" %in% colnames(maf))
+        stop("No \"hot_spot\" column in maf file. Annotate your maf file with GAMBLR::annotate_hot_spots() first. ")
       maf$Hugo_Symbol = ifelse(!is.na(maf$hot_spot), paste0(maf$Hugo_Symbol, "_hotspot"), maf$Hugo_Symbol)
     }
     #Convert the maf file to a binary matrix
     mutmat = maf %>%
       dplyr::select(Hugo_Symbol, Tumor_Sample_Barcode) %>%
-      left_join(dplyr::select(metadata, Tumor_Sample_Barcode, comparison), by = "Tumor_Sample_Barcode") %>%
+      full_join(dplyr::select(metadata, Tumor_Sample_Barcode, comparison), by = "Tumor_Sample_Barcode") %>%
       distinct() %>%
       dplyr::mutate(is_mutated = 1) %>%
       pivot_wider(names_from = Hugo_Symbol, values_from = is_mutated, values_fill = 0) %>%
       dplyr::mutate(across(where(is.numeric), ~replace_na(., 0)))
+
+    if(rm_na_samples){
```

Comment:
I don't think I understand how this works. What's the format of the table you're working with here? I can't follow it from the code.

Reply:
So, instead of a left join, mutmaf is now obtained with a full join (NA column retained). That temporary column is then renamed, and all rows that have 0 in that column are filtered out and the column is dropped (if the new parameter is set to true).  That being said, I am not 100% sure I understand what the issue is, so it might very well not be solving the issue that was raised.

Reply:
To test this you would need to supply a metadata table that has more samples in it than your MAF does (in at least one of the two groups you want to compare). The two ways of running it would be:
default (not implemented in original version: use total samples in the metadata for that group as your denominator rather than the subset that are in the MAF
optional (actual current behaviour): use the subset in the MAF as the denominator and ignore the fact that there are more cases in your group in the metadata

Reply:
Ok, thanks for explaining this a bit better. I will test it out and make some changes as needed.

Reply:
Since she wrote this function, @lkhilton can probably offer you a better review here than I if you get stuck.